main.go

```diff
@@ -22,59 +29,178 @@ func main() {
 		os.Exit(1)
 	}
 
-	// Collect Go Track1 function declarations that returns
-	pwd, _ := os.Getwd()
 	for _, pkg := range pkgs {
-		for _, f := range pkg.Syntax {
-			ast.Inspect(f, func(node ast.Node) bool {
-				asnmt, ok := node.(*ast.AssignStmt)
-				if !ok {
-					return true
-				}
-				if len(asnmt.Lhs) != 2 || len(asnmt.Rhs) != 1 {
-					return true
-				}
+		trackOneSDKScan(pkg, pwd)
+		pandoraSDKScan(pkg, pwd)
+	}
+}
 
-				ret1, ok := asnmt.Lhs[0].(*ast.Ident)
-				if !ok {
-					return true
-				}
-				if ret1.Name != "_" {
-					return true
-				}
+func trackOneSDKScan(pkg *packages.Package, rootPath string) {
+	for _, f := range pkg.Syntax {
+		ast.Inspect(f, func(node ast.Node) bool {
+			asnmt, ok := node.(*ast.AssignStmt)
+			if !ok {
+				return true
+			}
 
-				callExpr, ok := asnmt.Rhs[0].(*ast.CallExpr)
-				if !ok {
-					return true
-				}
-				selExpr, ok := callExpr.Fun.(*ast.SelectorExpr)
-				if !ok {
-					return true
-				}
-				funcObj := pkg.TypesInfo.Uses[selExpr.Sel]
-				signature, ok := funcObj.Type().(*types.Signature)
-				if !ok {
-					return true
-				}
-				res, ok := signature.Results().At(0).Type().(*types.Named)
-				if !ok {
-					return true
-				}
-				ures, ok := res.Underlying().(*types.Struct)
-				if !ok {
-					return true
-				}
-				futureType, ok := ures.Field(0).Type().(*types.Named)
-				if !ok {
-					return true
+			if len(asnmt.Lhs) != 2 || len(asnmt.Rhs) != 1 {
+				return true
+			}
+
+			ret1, ok := asnmt.Lhs[0].(*ast.Ident)
+			if !ok {
+				return true
+			}
+
+			if ret1.Name != "_" {
+				return true
+			}
+
+			callExpr, ok := asnmt.Rhs[0].(*ast.CallExpr)
+			if !ok {
+				return true
+			}
+			selExpr, ok := callExpr.Fun.(*ast.SelectorExpr)
+			if !ok {
+				return true
+			}
+
+			if pkg.TypesInfo == nil {
+				return true
+			}
+			funcObj, ok := pkg.TypesInfo.Uses[selExpr.Sel]
+			if !ok {
+				return true
+			}
+
+			signature, ok := funcObj.Type().(*types.Signature)
+			if !ok {
+				return true
+			}
+			res, ok := signature.Results().At(0).Type().(*types.Named)
+			if !ok {
+				return true
+			}
+			ures, ok := res.Underlying().(*types.Struct)
+			if !ok {
+				return true
+			}
+			futureType, ok := ures.Field(0).Type().(*types.Named)
+			if !ok {
+				return true
+			}
+			future := futureType.Obj()
+			if future.Name() != "FutureAPI" || future.Pkg().Path() != "github.com/Azure/go-autorest/autorest/azure" {
+				return true
+			}
+
+			fmt.Println("Track 1 Hit: " + strings.TrimPrefix(pkg.Fset.Position(asnmt.Pos()).String(), rootPath+string(filepath.Separator)))
+
+			return false
+		})
+	}
+}
+
+func pandoraSDKScan(pkg *packages.Package, rootPath string) {
```

Comment:
```suggestion
func pandoraSDKScan(pkg *packages.Package, pwd string) {
```

Can we rename it to be `pwd` to make it clear?